relay/messenger.go

```
@@ -0,0 +1,297 @@
+				if hexEncryptionKey != "" {
+					decodedBody, err := base64.StdEncoding.DecodeString(message.Body)
+					if err != nil {
+						logging.Logger.WithFields(logrus.Fields{
+							"session": session,
+							"key":     key,
+							"hash":    message.Hash,
+							// "body":    message.Body,
+							"error": err,
+						}).Error("Failed to decode data")
+						continue
+					}
+
+					decryptedBody, err = decrypt(string(decodedBody), hexEncryptionKey)
+					if err != nil {
+						logging.Logger.WithFields(logrus.Fields{
+							"session": session,
+							"key":     key,
+							"hash":    message.Hash,
+							// "body":    message.Body,
+							"error": err,
+						}).Error("Failed to decrypt data")
+						continue
+					}
+				}
+
+				if err := tssServerImp.ApplyData(decryptedBody); err != nil {
+					logging.Logger.WithFields(logrus.Fields{
+						"session": session,
+						"key":     key,
+						"error":   err,
+					}).Error("Failed to apply data")
+					continue
+				}
+
+				messageCache.Store(cacheKey, true)
+				client := http.Client{}
+				req, err := http.NewRequest(http.MethodDelete, server+"/message/"+session+"/"+key+"/"+message.Hash, nil)
+				if err != nil {
+					logging.Logger.WithFields(logrus.Fields{
+						"session": session,
+						"key":     key,
+						"error":   err,
+					}).Error("Failed to delete message")
+					continue
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					logging.Logger.WithFields(logrus.Fields{
+						"session": session,
+						"key":     key,
+						"error":   err,
+					}).Error("Failed to delete message")
+					continue
+				}
+
+				if resp.StatusCode != http.StatusOK {
+					logging.Logger.WithFields(logrus.Fields{
+						"session": session,
+						"key":     key,
+					}).Error("Failed to delete message, status code is not 200 OK")
+					continue
+				}
+			}
+		}
+	}
+}
+
+func encrypt(plainText, hexKey string) (string, error) {
+	key, _ := hex.DecodeString(hexKey)
```

don't swallow error , handle it appropriately